lib/modulesync.rb

```diff
@@ -173,8 +175,13 @@ def self.update(cli_options)
     managed_modules.each do |puppet_module|
       begin
         manage_module(puppet_module, module_files, defaults)
-      rescue # rubocop:disable Lint/RescueWithoutErrorClass
-        warn "Error while updating '#{puppet_module.given_name}'"
+      rescue ModuleSync::Error, Git::GitExecuteError => e
+        message = e.message || "Error during '#{options[:command]}'"
```

Comment:
You can consider wrapping `Git::GitExecuteError` in `ModuleSync::GitError` (which inherits from `ModuleSync::Error` inside `manage_module`.

Reply:
I cant see the benefit, instead, if we want to improve the exception usage, I think we should catch them at lower level (`repository.rb`) and throw an `ModuleSync::Repository::Error` error or something like this which is more generic but not covered by the improvements I made here.

Reply:
Well, that's sort of what I meant. `ModuleSync::Repository::Error` sounds even better. It would properly encapsulate the lower level operations. Consider it for the future.

Reply:
OK. I wrote it in my todo file (so many thing I want to contribute before...)